File: src/baseDatos/Serializador.java
package baseDatos;

import java.io.FileOutputStream;
import java.io.ObjectOutputStream;
import java.io.IOException;
import java.io.FileNotFoundException;
import java.util.List;

import gestorAplicacion.clasesPrincipales.*;

//Serializar todos los objetos creados durante la ejecucion del proyecto
 
public class Serializador {
    /*
     * lista <E>    Una lista de objetos. El generico se usa para poder agredar las clases que se crearon
     * className    El nombre de la clase que queremos usar como nombre del archivo
     */
    public static <E> void serializar(List<E> lista, String className) {

        try {
            String path = System.getProperty("user.dir") + "/baseDatos/temp/" + className + ".txt"; //Se borra /src ya que estaba quedando duplicado en la ruta
            FileOutputStream fileOut = new FileOutputStream(path); //donde serializar los archivos
            ObjectOutputStream objOut = new ObjectOutputStream(fileOut); //escribir en el archivo
            objOut.writeObject(lista);  // Guardamos la lista de objetos
            objOut.close();
            fileOut.close();
        } catch (FileNotFoundException e) {
            System.out.println("No se encuentra archivo");
        } catch (IOException e) {
            System.out.println("Error flujo de inicialización");
        }
        
    }
    
    //Las clases que se deben serializar
    public static void serializarTodo() {
        Serializador.serializar(Bus.getBuses(), "buses");
        Serializador.serializar(Conductor.getConductores(), "conductores");
        Serializador.serializar(Empresa.getEmpresas(), "empresas");
        Serializador.serializar(Pasajero.getPasajeros(), "pasajeros");
        Serializador.serializar(Ruta.getRutas(), "rutas");
        Serializador.serializar(Tiquete.getTiquetes(), "tiquetes");
        Serializador.serializar(Viaje.getViajes(), "viajes");
    }
}

File: Python/baseDatos/serializador.py
import pickle
from gestor_aplicacion.clasesPrincipales.Tiquete import Tiquete
from gestor_aplicacion.clasesPrincipales.Bus import Bus
from gestor_aplicacion.clasesPrincipales.Pasajero import Pasajero
from gestor_aplicacion.clasesPrincipales.Ruta import Ruta
from gestor_aplicacion.clasesPrincipales.Viaje import Viaje
from gestor_aplicacion.clasesPrincipales.Conductor import Conductor
import pathlib
import os


class Serializador:
    @classmethod
    def serializar(cls,tipo):
        if tipo=="Bus":
            pickfile = open("baseDatos/temp/buses.txt", "wb")
            serializa = Bus._buses
            pickle.dump(serializa, pickfile,)
            pickfile.close

        elif tipo=="Conductor":
            pickfile = open("baseDatos/temp/conductores.txt", "wb")
            serializa = Conductor._conductores
            pickle.dump(serializa, pickfile)
            pickfile.close
        
        elif tipo=="Pasajero":
            pickfile = open("baseDatos/temp/pasajeros.txt", "wb")
            serializa = Pasajero._pasajeros
            pickle.dump(serializa, pickfile)
            pickfile.close
        
        elif tipo=="Ruta":
            pickfile = open("baseDatos/temp/rutas.txt", "wb")
            serializa = Ruta._rutas
            pickle.dump(serializa, pickfile)
            pickfile.close
        
        elif tipo=="Tiquete":
            pickfile = open("baseDatos/temp/tiquetes.txt", "wb")
            serializa = Tiquete._tiquetes
            pickle.dump(serializa, pickfile)
            pickfile.close

        elif tipo == "Viaje":
            pickfile = open("baseDatos/temp/viajes.txt","wb")
            serializa = Viaje.getViajes()
            pickle.dump(serializa,pickfile)
            pickfile.close
        
    def SerializarTodo():
        Serializador.serializar("Bus")
        Serializador.serializar("Conductor")
        Serializador.serializar("Pasajero")
        Serializador.serializar("Ruta")
        Serializador.serializar("Tiquete")
        Serializador.serializar("Viaje")

